### PR TITLE
Support `wp language theme uninstall --all`

### PR DESCRIPTION
**Pull request by atoomic**
Refs: #122

Add support for '--all' to the Theme
uninstall function.

Switch the report output to use a table
and support for json.

Related https://github.com/wp-cli/wp-cli/issues/5832

### CODE REVIEW

**Comment by atoomic** [comment]
Install example:
```
[wpaccount@sandbox-c8-fat ~]$ ~/ww language theme install --all fr_FR
Using cached file '/home/wpaccount/.wp-cli/cache/translation/theme-twentytwentyone-1.0-fr_FR-1647106647.zip'...
Unpacking the update...
Installing the latest version...
Removing the old version of the translation...
Translation updated successfully.
Language 'fr_FR' for 'Twenty Twenty-One' installed.
Using cached file '/home/wpaccount/.wp-cli/cache/translation/theme-twentytwentythree-1.2-fr_FR-1667221284.zip'...
Unpacking the update...
Installing the latest version...
Removing the old version of the translation...
Translation updated successfully.
Language 'fr_FR' for 'Twenty Twenty-Three' installed.
Using cached file '/home/wpaccount/.wp-cli/cache/translation/theme-twentytwentytwo-1.4-fr_FR-1680121959.zip'...
Unpacking the update...
Installing the latest version...
Removing the old version of the translation...
Translation updated successfully.
Language 'fr_FR' for 'Twenty Twenty-Two' installed.
Warning: Language 'fr_FR' not available.
Language 'fr_FR' for 'UniBlock' not installed.
Success: Installed 3 of 4 languages (1 skipped).
+-------------------+--------+---------------+
| name              | locale | status        |
+-------------------+--------+---------------+
| twentytwentyone   | fr_FR  | installed     |
| twentytwentythree | fr_FR  | installed     |
| twentytwentytwo   | fr_FR  | installed     |
| uniblock          | fr_FR  | not available |
+-------------------+--------+---------------+
```

Uninstall example:
```
> wp language theme uninstall --all fr_FR
Language 'fr_FR' for 'twentytwentyone' uninstalled.
Language 'fr_FR' for 'twentytwentythree' uninstalled.
Language 'fr_FR' for 'twentytwentytwo' uninstalled.
Language fr_FR not installed.
Success: Language already uninstalled.
+-------------------+--------+---------------+
| name              | locale | status        |
+-------------------+--------+---------------+
| twentytwentyone   | fr_FR  | uninstalled   |
| twentytwentythree | fr_FR  | uninstalled   |
| twentytwentytwo   | fr_FR  | uninstalled   |
| uniblock          | fr_FR  | not installed |
+-------------------+--------+---------------+
```

File: src/Theme_Language_Command.php [hunk 1]
```diff
@@ -404,26 +444,87 @@ public function uninstall( $args, $assoc_args ) {
...
+
+				$results[] = (object) $result;
 			}
 		}
+
+		if ( 'summary' !== $assoc_args['format'] ) {
+			\WP_CLI\Utils\format_items( $assoc_args['format'], $results, array( 'name', 'locale', 'status' ) );
+		}
+
+		\WP_CLI\Utils\report_batch_operation_results( 'language', 'uninstall', $count, $successes, $errors, $skips );
```
[the remark this answers is not in the document]
Comment (atoomic) [reply src/Theme_Language_Command.php:527]: should do the trick 
```
$count = count( $themes ) * count( $language_codes );
```

**Comment by atoomic** [comment]
note: need to adjust and add some unit tests 
the `uninstall` is now using a table output

**Comment by atoomic** [comment]
fixed phpcs

**Comment by atoomic** [comment]
fixed and added some tests

**Review by swissspidy — APPROVED** [review]
Really nice work, thank you!

Would you also be open to add the same for plugins next? :-)

File: features/language-theme.feature [hunk 2]
```diff
@@ -269,3 +272,23 @@ Feature: Manage translation files for a WordPress install
...
+
+    When I run `wp language theme uninstall --all de_DE --format=csv`
+    Then the return code should be 0
+    And STDOUT should be:
+      """
+      name,locale,status
+      twentyseventeen,de_DE,"not installed"
+      twentysixteen,de_DE,"not installed"
+      """
+    And STDERR should be empty
```
Comment (danielbachhuber) [inline features/language-theme.feature:294]: Can we add a scenario for neither `<theme>` nor `--all` is provided?

File: src/Theme_Language_Command.php [hunk 3]
```diff
@@ -393,7 +406,33 @@ public function uninstall( $args, $assoc_args ) {
...
+		if ( empty( $assoc_args['format'] ) ) {
+			$assoc_args['format'] = 'table';
+		}
+
+		if ( in_array( $assoc_args['format'], array( 'json', 'csv' ), true ) ) {
+			$logger = new \WP_CLI\Loggers\Quiet();
+			\WP_CLI::set_logger( $logger );
+		}
+
+		/* process all themes */
```
Comment (danielbachhuber) [inline src/Theme_Language_Command.php:418]: This comment is extraneous and can be removed.

**Review by danielbachhuber — CHANGES_REQUESTED** [review]

**Review by danielbachhuber — APPROVED** [review]

**Pull request merged**